### PR TITLE
Refactor cavity2d code

### PR DESCRIPTION
**Pull request by yungyuc**
A quick example for (i) [adding a makefile](https://github.com/yungyuc/r_thingol/commit/4ba887b6aba39efb2ab6c3d14ad67ae32cb4ba35) for a Fortran code and (ii) [reindenting the source code](https://github.com/yungyuc/r_thingol/commit/eff539b9f9430ed8fb230ad137c9730c22700cb3).

The new, consistent indentation rules are (i) the starting indent is 6 characters, (ii) each indent level uses 3 characteres, and (iii) no tab is allowed.

### CODE REVIEW

File: code/cavity2d/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+*.mod
```
Comment (yungyuc) [inline code/cavity2d/.gitignore:1]: Add ignore file.

File: code/cavity2d/CAVITY_2D_STAGGERED.F90 [hunk 2]
```diff
@@ -9,499 +9,556 @@
...
 !!!!!!!! PLOT:  UY, VX
 !=============================================================================
-MODULE PISO
-parameter(MI=512,MJ=512)
-REAL XL,DT,DTO,DX,DXO,DY,DYO,DZ,DZO,CFL
-INTEGER NX,NXP,NXPP,NXM,NY,NYP,NYPP,NYM,NXH
-REAL GAM,GAMO,GAMM,GAMMO
-REAL DENI,UIN,VIN,WIN,PREI,RMACH,ENGI,RTI,REO
-END MODULE
+      MODULE PISO
```
Comment (yungyuc) [inline code/cavity2d/CAVITY_2D_STAGGERED.F90:12]: Indentation rules: (i) the starting indent is 6 characters, (ii) each indent level uses 3 characteres, and (iii) no tab is allowed.

File: code/cavity2d/CAVITY_2D_STAGGERED.F90 [hunk 3]
```diff
@@ -9,499 +9,556 @@
...
 
-	  ESLON=0.0000001; AV=0. ! AV: ARTIFICIAL VISCIUS IN FIRST VELOCITY PREDICATOR
+         ESLON=0.0000001; AV=0. ! AV: ARTIFICIAL VISCIUS IN FIRST VELOCITY PREDICATOR
 
-	  DO I=0,MIP; DO J=0,MJP
-	  X(I,J)=(I-0.5)*DX
-	  Y(I,J)=(J-0.5)*DY
-	  ENDDO; ENDDO
+         DO I=0,MIP
+            DO J=0,MJP
```
Comment (yungyuc) [inline code/cavity2d/CAVITY_2D_STAGGERED.F90:70]: Each do uses its own line.

File: code/cavity2d/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# How to run `CAVITY_2D_STAGGERED.F90`
```
Comment (yungyuc) [inline code/cavity2d/README.md:1]: Use markdown.

File: code/cavity2d/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+           2   1.53592497E-04   1.01379249E-02   6.67265477E-03   4.73801903E-02
+           ...
+```
+
+## Files included
+
+* `CAVITY_2D_STAGGERED.F90`: Fortran source code
+* `u-y_re100.png`: Comparison the Ghia et al. (1982) and PISO Code of
+  u-velocity along the vertical centreline through geometric center for
+  Re=100.  ![u-y RE=100](u-y_re100.png)
```
Comment (yungyuc) [inline code/cavity2d/README.md:27]: Show the picture directly on github.

**Comment by yungyuc** [comment]
@markshake Please take a look at this PR as an example for how to use github to collaboratively develop code.  Note that each check-in is a standalone step that serves its own purpose.

**Comment by markshake** [comment]
Thanks a lot.
I will carefully study the differences.

**Pull request merged**